Makefile

```diff
@@ -248,11 +248,11 @@ lint-modified:     		  ## Run code linter to check code style, check if formatte
 check-aws-markers:     		  ## Lightweight check to ensure all AWS tests have proper compatibilty markers set
 	($(VENV_RUN); python -m pytest --co tests/aws/)
 
-format:            		  ## Run ruff and black to format the whole codebase
-	($(VENV_RUN); python -m ruff check --output-format=full --fix .; python -m black .)
+format:            		  ## Run ruff to format the whole codebase
+	($(VENV_RUN); python -m ruff format .; python -m ruff check --output-format=full --fix .)
```

Comment:
If we only want to sort imports here with the linter, we can do only:

`ruff check --select I --fix`

Reply:
Actually, this is the case for a while now (and not really in the scope of this PR)... I guess I would also say that I expect the formatter changing the code by applying linting fixes? But that's definitely arguable...

Reply:
The formatter itself sadly can't sort imports:

`https://docs.astral.sh/ruff/formatter/#sorting-imports` 

I'd say linting goes deeper than formatting, so the formatter shouldn't apply those fixes.

Reply:
This change here only replaces the usage of `black` with `ruff format`.
I agree that it might be arguable if the `format` make target should do that, but honestly I don't think this is in the scope of this PR.
Happy to discuss this in a follow-up PR from you as soon as this is through ;)